cmd/bucky/common_sync.go

```diff
@@ -0,0 +1,342 @@
+	c := 0
+	for _, jobss := range jobsd {
+		for _, jobs := range jobss {
+			c = c + len(jobs)
+		}
+	}
+	return c
+}
+
+func (ms *metricSyncer) sync(jobc chan *syncJob, srcThrottling map[string]chan struct{}, wg *sync.WaitGroup) {
+	for job := range jobc {
+		func() {
+			src, dst := job.srcServer, job.dstServer
+
+			srcThrottling[src] <- struct{}{}
+
+			syncStart := time.Now()
+			defer func() {
+				atomic.AddInt64(&ms.stat.time.count, 1)
+				atomic.AddInt64(&ms.stat.time.total, int64(time.Since(syncStart)))
+
+				<-srcThrottling[src]
+			}()
+
+			if ms.flags.verbose {
+				log.Printf("Relocating [%s] %s => [%s] %s  Delete Source: %t",
+					src, job.oldName,
+					dst, job.newName, ms.flags.delete)
+			}
+
+			var mhstats *metricHealStats
+			if ms.flags.offloadFetch {
+				var err error
+				mhstats, err = CopyMetric(src, dst, job.oldName)
+				if err != nil {
+					// errors already loggged in the func
+					if errors.Is(err, errNotFound) {
+						ms.stat.notFound++
+					} else {
+						ms.stat.copyError++
+					}
+
+					return
+				}
+			} else {
+				metric, err := GetMetricData(src, job.oldName)
+				if err != nil {
+					// errors already loggged in the func
+					if errors.Is(err, errNotFound) {
+						ms.stat.notFound++
+					} else {
+						ms.stat.copyError++
+					}
+
+					return
+				}
+				metric.Name = job.newName
+				mhstats, err = PostMetric(dst, metric)
+				if err != nil {
+					// errors already loggged in the func
+					if errors.Is(err, errNotFound) {
+						ms.stat.notFound++
+					} else {
+						ms.stat.copyError++
+					}
+
+					return
+				}
+			}
+
+			if mhstats != nil {
+				if mhstats.Download > 0 {
+					atomic.AddInt64(&ms.stat.time.download.count, 1)
+					atomic.AddInt64(&ms.stat.time.download.total, mhstats.Download)
+				}
+				if mhstats.Dump > 0 {
+					atomic.AddInt64(&ms.stat.time.dump.count, 1)
+					atomic.AddInt64(&ms.stat.time.dump.total, mhstats.Dump)
+				}
+				if mhstats.Fill > 0 {
+					atomic.AddInt64(&ms.stat.time.fill.count, 1)
+					atomic.AddInt64(&ms.stat.time.fill.total, mhstats.Fill)
+				}
+				if mhstats.Compress > 0 {
+					atomic.AddInt64(&ms.stat.time.compress.count, 1)
+					atomic.AddInt64(&ms.stat.time.compress.total, mhstats.Compress)
+				}
+				if mhstats.Copy > 0 {
+					atomic.AddInt64(&ms.stat.time.copy.count, 1)
+					atomic.AddInt64(&ms.stat.time.copy.total, mhstats.Copy)
+				}
+			}
+
+			// We only delete if there are no errors present
+			if ms.flags.delete {
+				deleteStart := time.Now()
+
+				err := DeleteMetric(src, job.oldName)
+				if err != nil {
+					// errors already loggged in the func
+					ms.stat.deleteError++
+				}
+
+				atomic.AddInt64(&ms.stat.time.delete.count, 1)
+				atomic.AddInt64(&ms.stat.time.delete.total, int64(time.Since(deleteStart)))
+			}
+		}()
+	}
+
+	wg.Done()
+}
+
+func (ms *metricSyncer) displayProgressOrExit(total int, progress *int64, progressDone chan struct{}) {
+	var start = time.Now().Unix()
```

Comment:
It's better to have variable declarations consistent across the whole change. E.x. if you use `:=` then it's better to stick to it everywhere, with exception of when you want to have default value (like on line 283 in this file).

Reply:
Hi @Civil , for this case, please allow me to keep it the way it is. I just prefer to keep the whole block a bit consistent by  using var. It looks neater to me, instead of mixing `var` and `:=`

```go
var start = time.Now().Unix()
var ptime = time.Now().Unix()
var pprogress int64
var exit, forceExit bool
var ticker = time.NewTicker(time.Second * 10)
```